LibNFS.xs

```diff
@@ -983,6 +983,33 @@ static void _set_unix_authn(pTHX_ struct nfs_context* nfs, SV* value_sv) {
     nfs_set_auth(nfs, auth);
 }
 
+#ifdef NLNFS_NFS_SET_READDIR_MAX_BUFFER_SIZE
+static void _set_readdir_max_buffer_size(pTHX_ struct nfs_context* nfs, SV* value_sv) {
+
+    if (SvROK(value_sv) && (SvTYPE(SvRV(value_sv)) == SVt_PVAV)) {
+        AV* av = (AV*) SvRV(value_sv);
+        uint32_t nums_count = 1 + av_len(av);
+        if (nums_count != 2) croak("“%s” must contain exactly 2 numbers", "readdir_buffer");
+
+        uint32_t nums[nums_count];
+
+        for (unsigned n=0; n<nums_count; n++) {
+            SV** svp = av_fetch(av, n, 0);
+            assert(svp);
+
+            nums[n] = exs_SvUV(*svp);
```

Comment:
Should reject if `exs_SvUV(*svp) > UINT32_MAX` before assigning into `nums[n]`.

Reply:
Same here. Will adjust this in _set_unix_authn() and use uint64_t here as upstream. Do I still need a check then?

Reply:
I wouldn’t say `_set_unix_authn()` needs a change here, though if there’s a disparity in int sizes there then that should be addressed separately.

If upstream takes a u32, then we need to ensure that our UV doesn’t exceed what a u32 can hold.

Reply:
I have done various tests with UID 1_000_000_000, 4_000_000_000 (> INT32_MAX but < UINT32_MAX) and 5_000_000_000 (> UINT32_MAX). While the first two worked fine and survived the implicit casts back and forth, 5_000_000_000 appeared as 705_032_704, i.e. only the lower 4 bytes were considered. So I created #8 to restrict the values to UINT32_MAX.

Reply:
> Should reject if `exs_SvUV(*svp) > UINT32_MAX` before assigning into `nums[n]`.

Done.

Reply:
While I was checking in Wireshark, I noticed that the field is also just 4 bytes. I somehow misinterpreted the type definitions. I'll send a PR for libnfs and will also ask about nfs_set_uid().